### PR TITLE
Scrim rejection backend and frontend

### PR DESCRIPTION
**Pull request by zachstence**
- Write scrim rejection mutation and query
- Modify submission service to allow resubmission after rejection
- Audit rejections on submission in redis, with a reason
- Analytics on rejection with `playerId` and `submissionId`

### CODE REVIEW

**Comment by 0c370t** [comment]
Task linked: [CU-2cdm5gh Scrim rejection backend](https://app.clickup.com/t/2cdm5gh)

File: clients/web/src/lib/components/organisms/scrims/QueuedSubViews/RatificationView.svelte [hunk 1]
```diff
@@ -2,11 +2,13 @@
     import type {CurrentScrim} from "$lib/api";
     import {RatifySubmissionMutation, SubmissionStatsStore} from "$lib/api";
     import {GameCard, Progress} from "$lib/components";
+import RejectSubmissionModal from "../modals/RejectSubmissionModal.svelte";
```
Comment (bwpikaard) [inline clients/web/src/lib/components/organisms/scrims/QueuedSubViews/RatificationView.svelte:5]: Spacing, lint

**Review by bwpikaard — APPROVED** [review]

File: clients/web/src/lib/components/organisms/scrims/QueuedSubViews/RatificationView.svelte (hunk 1 above)
Comment (0c370t) [inline clients/web/src/lib/components/organisms/scrims/QueuedSubViews/RatificationView.svelte:5]: Format pls

File: common/src/service-connectors/matchmaking/matchmaking.types.ts [hunk 2]
```diff
@@ -75,6 +76,10 @@ export const MatchmakingSchemas = {
         input: Schemas.GetScrimBySubmissionId_Request,
         output: Schemas.GetScrimBySubmissionId_Response,
     },
+    [MatchmakingEndpoint.ResetScrim]: {
+        input: Schemas.ResetScrim_Request,
+        output: Schemas.ResetScrim_Response,
+    },
```
Comment (0c370t) [inline common/src/service-connectors/matchmaking/matchmaking.types.ts:82]: Can we use `MatchmakingEndpoint.ForceUpdateScrimStatus` instead?

Comment (0c370t) [reply common/src/service-connectors/matchmaking/matchmaking.types.ts:82]: At a cursory glance it looks like we can, would be better to reuse that

**Review by 0c370t — CHANGES_REQUESTED** [review]

File: clients/web/src/lib/components/organisms/scrims/QueuedSubViews/RatificationView.svelte (hunk 1 above)
Comment (zachstence) [reply clients/web/src/lib/components/organisms/scrims/QueuedSubViews/RatificationView.svelte]: https://github.com/SprocketBot/sprocket/pull/106/commits/a8a0ca4a39650b43289711347ecbf692125cb9dc

Comment (zachstence) [reply clients/web/src/lib/components/organisms/scrims/QueuedSubViews/RatificationView.svelte]: https://github.com/SprocketBot/sprocket/pull/106/commits/a8a0ca4a39650b43289711347ecbf692125cb9dc

File: common/src/service-connectors/matchmaking/matchmaking.types.ts (hunk 2 above)
Comment (zachstence) [reply common/src/service-connectors/matchmaking/matchmaking.types.ts]: https://github.com/SprocketBot/sprocket/pull/106/commits/462640665da7583e04655e7d7cf7ed66b4f4051d

**Pull request merged**